src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusExporterMiddleware.cs

```diff
@@ -22,21 +22,22 @@ internal sealed class PrometheusExporterMiddleware
     /// </summary>
     /// <param name="meterProvider"><see cref="MeterProvider"/>.</param>
     /// <param name="next"><see cref="RequestDelegate"/>.</param>
-    public PrometheusExporterMiddleware(MeterProvider meterProvider, RequestDelegate next)
+    /// <exception cref="ArgumentException">Thrown when a <see cref="PrometheusExporter"/> could not be found on the provided <see cref="MeterProvider"/>.</exception>
+    public PrometheusExporterMiddleware(MeterProvider meterProvider, RequestDelegate? next)
     {
         Guard.ThrowIfNull(meterProvider);
 
-        if (!meterProvider.TryFindExporter(out PrometheusExporter exporter))
+        if (!meterProvider.TryFindExporter(out PrometheusExporter? exporter))
         {
             throw new ArgumentException("A PrometheusExporter could not be found configured on the provided MeterProvider.");
         }
 
-        this.exporter = exporter;
+        this.exporter = exporter ?? throw new ArgumentNullException(nameof(exporter));
```

Comment:
This is wrong.

`ArgumentNullException` is inappropriate here because "exporter" wasn't passed into this ctor.

The `TryFindExporter` method already specifies that `exporter` is not null when this method returns true, so this null check shouldn't be needed. Compiler might want `this.exporter` to be specified as nullable.